src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -554,10 +556,14 @@ private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher
     }
 
 
-    private String[] getPreRunnerSwitches() {
+    private String[] getPreRunnerSwitches() throws MatlabVersionNotFoundException {
         String[] preRunnerSwitches =
                 {getLocalMatlab() + nodeSpecificfileSeparator + "bin" + nodeSpecificfileSeparator + "matlab", "-nosplash",
-                        "-nodesktop", "-noAppIcon"};
+                        "-nodesktop"};
+        MatlabReleaseInfo rel = new MatlabReleaseInfo(this.nodeSpecificMatlabRoot);
```

Comment:
We are creating another MatlabReleaseInfo object here, whereas one was already created in the earlier method. This means that we will parse the same XML file twice. Since MatlabReleaseInfo is not changing we could store this as a class property.

Reply:
Done.